### PR TITLE
Generate release notes for 2.0.0

### PR DESCRIPTION
**Pull request by peternied**
### Description
How to repeat:
* `git checkout origin/2.0`
* `git log --pretty=format:"* %s" --since=3-11-2022 > release-notes/opensearch-security.release-notes-2.0.0.0-rc1.md`
* Manually organize file sections
  * Add Date / Version section (copy + modify from previous release)
  * Enhancements/Bug fixes/Maintenance
* Replace pull request ids with full urls via regex
  * Find: `\(#(\d+)\)$`
  * Replace: `([#$1](https://github.com/opensearch-project/security/pull/$1))`
* Commit with these steps / modified with additional details

### Check List
- [ ] ~New functionality includes testing~
- [ ] ~New functionality has been documented~
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-security.release-notes-2.0.0.0-rc1.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+## 2022-04-20 Version 2.0.0.0
+
+Compatible with OpenSearch 2.0.0-rc1
+
+### Enhancements
+
+* Incremented version to 2.0-rc1. ([#1764](https://github.com/opensearch-project/security/pull/#1764))
```
Comment (davidlago) [inline release-notes/opensearch-security.release-notes-2.0.0.0-rc1.md:7]: calling this an enhancement is a bit of a stretch :) this feels more of a maintenance issue.

File: release-notes/opensearch-security.release-notes-2.0.0.0-rc1.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+* Renames securityconfig folder to config in bundle step and makes relevant changes ([#1749](https://github.com/opensearch-project/security/pull/#1749))
+* Add signal/wait model for TestAuditlogImpl ([#1758](https://github.com/opensearch-project/security/pull/#1758))
+* Remove TransportClient auth/auth ([#1701](https://github.com/opensearch-project/security/pull/#1701))
+* Introduce dfm_empty_overrides_all setting to enable role without dls/fls to override roles with dls/fls ([#1735](https://github.com/opensearch-project/security/pull/#1735))
+* Add depreciation notice to security tools ([#1756](https://github.com/opensearch-project/security/pull/#1756))
+* [Practice] Reverting changes ([#1754](https://github.com/opensearch-project/security/pull/#1754))
```
Comment (davidlago) [inline release-notes/opensearch-security.release-notes-2.0.0.0-rc1.md:13]: These links are incorrect

**Review by davidlago — CHANGES_REQUESTED** [review]

File: release-notes/opensearch-security.release-notes-2.0.0.0-rc1.md (hunk 2 above)
Comment (cliu123) [reply release-notes/opensearch-security.release-notes-2.0.0.0-rc1.md:13]: All links should be `.../pull/<PR_number>`

File: release-notes/opensearch-security.release-notes-2.0.0.0-rc1.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+## 2022-04-20 Version 2.0.0.0
+
+Compatible with OpenSearch 2.0.0-rc1
+
+### Enhancements
+
+* Incremented version to 2.0-rc1. ([#1764](https://github.com/opensearch-project/security/pull/#1764))
+* Renames securityconfig folder to config in bundle step and makes relevant changes ([#1749](https://github.com/opensearch-project/security/pull/#1749))
```
Comment (cliu123) [inline release-notes/opensearch-security.release-notes-2.0.0.0-rc1.md:8]: This looks more of `Maintenance` too.

File: release-notes/opensearch-security.release-notes-2.0.0.0-rc1.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+* Add signal/wait model for TestAuditlogImpl ([#1758](https://github.com/opensearch-project/security/pull/#1758))
+* Remove TransportClient auth/auth ([#1701](https://github.com/opensearch-project/security/pull/#1701))
```
Comment (cliu123) [inline release-notes/opensearch-security.release-notes-2.0.0.0-rc1.md:10]: This is a `Maintenance`.

File: release-notes/opensearch-security.release-notes-2.0.0.0-rc1.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+* Introduce dfm_empty_overrides_all setting to enable role without dls/fls to override roles with dls/fls ([#1735](https://github.com/opensearch-project/security/pull/#1735))
+* Add depreciation notice to security tools ([#1756](https://github.com/opensearch-project/security/pull/#1756))
+* [Practice] Reverting changes ([#1754](https://github.com/opensearch-project/security/pull/#1754))
+* Upgrade to opensearch 2.0.0 alpha1 ([#1741](https://github.com/opensearch-project/security/pull/#1741))
```
Comment (cliu123) [inline release-notes/opensearch-security.release-notes-2.0.0.0-rc1.md:14]: #1741, #1590 and #1698 all should be `Maintenance`.

**Comment by cliu123** [comment]
Several PRs that are in 2.0 branch but not in 1.3 branch are missing, for example [#1698](https://github.com/opensearch-project/security/pull/1698)

**Comment by cliu123** [comment]
Several PRs that are in 2.0 branch but not in 1.3 branch are missing, for example https://github.com/opensearch-project/security/pull/1709

**Comment by cliu123** [comment]
Please include https://github.com/opensearch-project/security/pull/1775

File: release-notes/opensearch-security.release-notes-2.0.0.0-rc1.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
+## 2022-04-20 Version 2.0.0.0
```
Comment (cliu123) [inline release-notes/opensearch-security.release-notes-2.0.0.0-rc1.md:1]: The version is `2.0.0.0-rc1`

Comment (cliu123) [inline release-notes/opensearch-security.release-notes-2.0.0.0-rc1.md:1]:
```suggestion
## 2022-04-20 Version 2.0.0.0-rc1
```

File: release-notes/opensearch-security.release-notes-2.0.0.0-rc1.md (hunk 2 above)
Comment (peternied) [reply release-notes/opensearch-security.release-notes-2.0.0.0-rc1.md]: Thanks, fixed

File: release-notes/opensearch-security.release-notes-2.0.0.0-rc1.md (hunk 4 above)
Comment (peternied) [reply release-notes/opensearch-security.release-notes-2.0.0.0-rc1.md]: We do not have a firm definition for these terms, I'd like to consider as many items as possible to be enhancements as they improve how our build functions and codebase operates.

For your consideration `Renames securityconfig folder to config in bundle step and makes relevant changes ([#1749](https://github.com/opensearch-project/security/pull/#1749))` was done to improve user experience when browsing the plugin folder hierarchy.

**Comment by peternied** [comment]
> Several PRs that are in 2.0 branch but not in 1.3 branch are missing, for example https://github.com/opensearch-project/security/pull/1709

@cliu123  Thanks for checking on this, I wasn't going back far enough, pulling all changes since the last release notes was published

**Review by davidlago — APPROVED** [review]

**Review by DarshitChanpura — APPROVED** [review]

**Pull request merged**